### ISSUE TITLE
Navigation and Table of Contents widgets set the wrong items as active while scrolling if the height of the content changes

### ISSUE THREAD

**Issue by lime517** [elementor/elementor#11715]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

When using the Table of Contents Widget, if the height of the content changes, the highlighted/active item is wrong.

In my example, I have 5 headings, with varying lengths of content below each. Here is how it looks when I first load the page:

![image](https://user-images.githubusercontent.com/7725357/85408164-2ad7cf00-b532-11ea-828d-cfc129901dc6.png)

I expand some of the toggles in the content, and continue scrolling. We can see that 'Fifth Heading' is highlighted even though I can't see past the Third. Note that I've set the 'Table of Contents' widget as sticky.

![image](https://user-images.githubusercontent.com/7725357/85408381-71c5c480-b532-11ea-8429-059525b649b7.png)

Resizing the window fixes the issue, at least until the height changes again.

## Steps to reproduce

1. Create a section with plenty of content, multiple headings, and some Toggles (which we will use to change the height of the content)
2. Add in a Table of Contents widget. To see the issue better, I recommend placing it in a separate column, and setting it to sticky, so that you can always see it while scrolling.
3. Scroll up and down the page, you should see the active item in the Table of Contents widget change as you scroll up and down.
4. Toggle one of the toggles, changing the height of the content.
5. Scroll up and down the page, you should see that the wrong active items are set in the Table of Contents widget.
6. (Optional) If you trigger a window resize event, the issue is temporarily fixed.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
== Server Environment ==
	Operating System: Darwin
	Software: nginx/1.16.0
	MySQL version: MySQL Community Server - GPL v8.0.16
	PHP Version: 7.3.5
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.4.2
	Site URL: http://elementor-bug-test.local
	Home URL: http://elementor-bug-test.local
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Twenty Nineteen
	Version: 1.6
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 2.9.13
		Author: Elementor.com

	Elementor Pro
		Version: 2.10.2
		Author: Elementor.com


== Elements Usage ==
	
	wp-page : 1
		heading : 5
		table-of-contents : 1
		text-editor : 5
		toggle : 2


== Log ==
	
```
</details>

**Comment by iStuff0** [comment]
Yes, i came here wondering the same thing. Did you find a workaround? The highlight also seems to bounce around erroneously when scrolling down a page, particularly when sub items are collapsed. Shame really as an accurate highlight would be posh!

**Comment by lime517** [comment]
I built a _very_ messy Javsascript snippet that worked around it, but I honestly forget exactly what I did to circumvent the bug.

**Comment by davronakil** [comment]
Same issue happening here. Any tips to resolve this?

**Comment by sriramgroot** [comment]
This issue happening for me too

**Comment by LouiScura** [comment]
This worked for me at least looks less bad lol

`.elementor-menu-anchor {
    position: absolute;
    margin-top: calc(50vh);
    scroll-margin-top: calc(50vh + 50px) !important;
}`
